mbed_tools_ci_scripts/utils/third_party_licences.py

```diff
@@ -90,8 +92,19 @@ def get_licence(self, licence_descriptor: Optional[str]) -> Optional[Licence]:
         self.load()
         if not self._licence_store or not self._licence_list or not licence_descriptor:
             return None
-        # TODO refine the licence_descriptor: remove The at begining of strings.
-        return self._licence_store.get(determine_similar_string_from_list(licence_descriptor, self._licence_list))
+        cleansed_descriptor = cleanse_licence_descriptor(licence_descriptor)
+        likelihood, licence = determine_similar_string_from_list(cleansed_descriptor, self._licence_list)
```

Comment:
I'm wondering if it might be better to push the "likelihood" check down into `determine_similar_string_from_list`? That way we don't have to return a `tuple` and the function has a clearer post-condition, i.e it returns `None` if there's no similar string in the list, or it just returns the similar string. At the moment it's returning `candidate_similar_string_from_list` and passing the responsibility of determining whether that is a similar string or not to the caller. Thoughts?

Reply:
The reason I did it like that is that the threshold value is really dependent on the kind of strings we are dealing with. For licences, a threshold of 0.5 seems to be good enough. However, for other types of strings, this may be completely erroneous. As these helpers are meant to be generic enough so that they can be reused elsewhere, I did not want to make too bold decisions within the helper method but rather let the responsibility to make them where we have more context about the kind of strings we are dealing with